### PR TITLE
[readosm] Revise portfile, install pc file for windows

### PR DESCRIPTION
**Pull request by dg0yt**
- #### What does your PR fix?  
  Revises the portfile. Uses pkgconfig for nmake and autotools.
  Installs pc file also for windows. (Needed for #21594.)

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  all, yes

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  yes

### CODE REVIEW

File: scripts/ci.baseline.txt [hunk 1]
```diff
@@ -1663,8 +1663,8 @@ open62541:x64-windows-static-md=fail
 openscap:x64-windows-static-md=fail
 portmidi:x64-windows-static-md=fail
 quantlib:x64-windows-static-md=fail
-readosm:x64-windows-static-md=fail
 sentencepiece:x64-windows-static-md=fail
+spatialite-tools:x64-windows-static-md=fail
```
Comment (dg0yt) [inline scripts/ci.baseline.txt:1667]: To be revisited and probably fixed with #21594. (Used to be cascaded from the readosm fail entry.)

File: ports/readosm/portfile.cmake [hunk 2]
```diff
@@ -1,89 +1,90 @@
...
-    set(CL_FLAGS_DBG "/MDd /Zi")
-    set(CL_FLAGS_REL "/MD /Ox")
-    set(EXPAT_LIBS_REL "${CURRENT_INSTALLED_DIR}/lib/libexpat.lib")
-    set(EXPAT_LIBS_DBG "${CURRENT_INSTALLED_DIR}/debug/lib/libexpatd.lib")
-  else()
-    set(CL_FLAGS_DBG "/MTd /Zi")
-    set(CL_FLAGS_REL "/MT /Ox")
-    set(EXPAT_LIBS_REL "${CURRENT_INSTALLED_DIR}/lib/libexpatMD.lib")
-    set(EXPAT_LIBS_DBG "${CURRENT_INSTALLED_DIR}/debug/lib/libexpatdMD.lib")
-  endif()
```
Comment (JackBoosY) [inline ports/readosm/portfile.cmake:26]: Do we not need this now? Can you please check the build log?

**Review by JackBoosY — CHANGES_REQUESTED** [review]

**Comment by dg0yt** [comment]
Full integration succeeded in #21594. (readosm, libspatialite, spatialite-tools).

File: ports/readosm/portfile.cmake (hunk 2 above)
Comment (JackBoosY) [reply ports/readosm/portfile.cmake:26]: Alright, I will check this locally.

**Comment by JackBoosY** [comment]
I think we should pick PR #21594 instead of this.

**Comment by dg0yt** [comment]
> I think we should pick PR #21594 instead of this.

Fine for me.

**Pull request closed without merging**